src/digest/adler32.cr

```diff
@@ -19,4 +27,27 @@ module Digest::Adler32
   def self.combine(adler1 : UInt32, adler2 : UInt32, len) : UInt32
     LibZ.adler32_combine(adler1, adler2, len).to_u32
   end
+
+  # :nodoc:
+  def update_impl(data : Bytes) : Nil
+    @digest = Adler32.update(data, @digest)
+  end
+
+  # :nodoc:
+  def final_impl(dst : Bytes) : Nil
+    dst[0] = (@digest >> 24).to_u8!
+    dst[1] = (@digest >> 16).to_u8!
+    dst[2] = (@digest >> 8).to_u8!
+    dst[3] = (@digest).to_u8!
```

Comment:
There is a dedicated method for big endian encoding which we can use here.

```suggestion
    IO::ByteFormat::BigEndian.encode(@digest, dst)
```

It's probably also little bit more efficient, using a single `memcpy`.

Reply:
```
[blobcodes@toolbox mangaba]$ crystal eval '
require "benchmark"

bytes_store = uninitialized UInt8[4]
bytes = bytes_store.to_slice
n = rand(UInt32::MAX)

Benchmark.ips do |x|
  x.report("after") { IO::ByteFormat::BigEndian.encode(n, bytes) }
  x.report("before") do
    bytes[0] = (n >> 24).to_u8!
    bytes[1] = (n >> 16).to_u8!
    bytes[2] = (n >> 8).to_u8!
    bytes[3] = (n).to_u8!
  end
end

# Avoid LLVM optimizations by actually using the result
puts bytes
' --release
 after 173.22M (  5.77ns) (± 0.64%)  0.0B/op   1.55× slower
before 269.07M (  3.72ns) (± 0.84%)  0.0B/op        fastest
Bytes[62, 20, 67, 3]
```

It is slower and the same pattern is also used in other places:

https://github.com/crystal-lang/crystal/blob/d72f6eaf7b42ab32d0ce3f9e64d09f0c0509e089/src/crypto/bcrypt.cr#L141-L144